python/scripts/openfisca_json_customizer.py

```diff
@@ -19,73 +19,96 @@
 
 import json
 import sys
+from collections import OrderedDict
 import dpath.util
+from dpath.exceptions import PathNotFound 
+import logging
 
 
 # Main functions
 
-def update_json(json_file):
+handler = logging.StreamHandler()
+log = logging.getLogger(__name__)
+log.addHandler(handler)
+log.setLevel(logging.DEBUG)
+
+
+def update_json(json_file, update_actions):
   with open(json_file, 'r+') as f:
-    json_content = json.load(f)
+    json_content = json.load(f, object_pairs_hook=OrderedDict)
     update_actions(json_content)
     update_file(f, json_content)
 
 
 def update_actions(json_content):
-  # E.g. : Mise à jour de la valeur d'id et vérification de succès
-  id_key = u'_id'
+  log.info("Add in 'update_actions' function any action you want to apply to json_content.")
+
+  # E.g. : Updating '_id' key value and priting it
+  # id_key = u'_id'
   # json_content = update_value(json_content, id_key, u'toto')
   # print_value(json_content, id_key)
 
-  # E.g. : Suppression d'id et vérification de succès
-  json_content = delete_key(json_content, id_key)
-  print_value(json_content, id_key)
-
 
 # Helpers
 
+def is_json(file_path):
+  return file_path[-5:] == ".json"
+
 
 def print_value(json_content, key):
   try:
-    print key + ": " + dpath.util.get(json_content, key)
+    log.info(key + u": " + str(dpath.util.get(json_content, key)))
   except KeyError as e:
-    print 'Clef ' + e.message + ' introuvable.'
+    log.error(e.message + u' key non found.')
 
 
 def print_json(json_content):
-  print json.dumps(json_content, indent=4, sort_keys=True)
+  log.info(json.dumps(json_content, indent=4, sort_keys=True))
 
 
 def add_key(json_content, new_key, new_value):
   dpath.util.new(json_content, new_key, new_value)
-  return json_content
 
 
 def update_value(json_content, key, new_value):
   dpath.util.set(json_content, key, new_value)
-  return json_content
 
 
 def update_key(json_content, old_key, new_key, new_value):
```

Comment:
Done by error to translate mes-aides `nationalite` to openfisca `ressortissant_eee` and kept to manage openfisca variables renaming through time.

Reply:
In the case of a rename, wouldn't it be more interesting to be able to change the key whitout changing the value ?
Because in the case of the update key, the user has to give "new value" as an argument, where in the rename usecase, that value would stay the same.